### PR TITLE
Add support for sliced/gzipped input

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-294

Changes:
- The processor only supported a simple CSV file at the input, which was then converted to a sliced/gzipped file.
- Now the processor supports sliced/gzipped files also on the input.

### CODE REVIEW

File: test/processor/sliced-table-without-columns/expected-stderr [hunk 1]
```diff
@@ -0,0 +1 @@
+Error: the manifest "*/in/tables/table.csv.manifest" has no columns, columns are required for the sliced table
```
Comment (michaljurecko) [inline test/processor/sliced-table-without-columns/expected-stderr:1]: Pridany E2E test.

File: test/processor/sliced-table-without-manifest/expected-stderr [hunk 2]
```diff
@@ -0,0 +1 @@
+Error: the manifest "*/in/tables/table.csv.manifest" not found, it is required for the sliced table
```
Comment (michaljurecko) [inline test/processor/sliced-table-without-manifest/expected-stderr:1]: Pridany E2E test.

File: test/processor/datadir_test.go [hunk 3]
```diff
@@ -63,6 +63,9 @@ func RunDataDirTest(t *testing.T, testDir string, binary string) {
 		}
 	}
 
+	// Gzip files for easier definition
+	gzipAllInDir(t, dataDir+"/in")
+
```
Comment (michaljurecko) [inline test/processor/datadir_test.go:68]: V datadir testoch je mozne v `out` - expected dir vytvorit subory s ext `.ungzipped`.
Pouziva sa to ako procesor vytvara gzip subory, tak aby v git repe boli v dekodovanom formate.

V tomto commite som to pridal aj do `in` ale opacne, ze sa tieto subory pred testom prevedu do `gzip`.

File: test/processor/gzip-in/expected-stdout [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Configured max 20 KiB per slice.
+Slicing table "tables/tenRows.csv".
+Table "tables/tenRows.csv" sliced: in/out: 6 / 1 slices, *B / 101B bytes, 10 rows, manifest unaffected.
```
Comment (michaljurecko) [inline test/processor/gzip-in/expected-stdout:3]: Pridany E2E test `gzip-in`.

File: test/processor/gzip-in-out/expected-stdout [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Configured max 30 B per slice.
+Gzip enabled, compression level = 2.
+Slicing table "tables/tenRows.csv".
+Table "tables/tenRows.csv" sliced: in/out: 6 / 4 slices, *B / *B bytes, 10 rows, manifest unaffected.
```
Comment (michaljurecko) [inline test/processor/gzip-in-out/expected-stdout:4]: Pridany E2E test `gzip-in-out`.

File: test/processor/sliced-table/expected-stdout [hunk 6]
```diff
@@ -0,0 +1,3 @@
+Configured max 30 B per slice.
+Slicing table "tables/tenRows.csv".
+Table "tables/tenRows.csv" sliced: in/out: 6 / 4 slices, 101B / 101B bytes, 10 rows, manifest unaffected.
```
Comment (michaljurecko) [inline test/processor/sliced-table/expected-stdout:3]: Pridany E2E test `sliced-table`.

File: internal/pkg/slicer/rowsreader/reader.go [hunk 7]
```diff
@@ -15,77 +20,135 @@ const (
...
+
+	// Stream slices to the pipe
+	pipeReader, pipeWriter := io.Pipe()
+	go func() {
+		// Iterate all slices
+		err := filepath.WalkDir(dirPath, func(path string, d fs.DirEntry, walkErr error) (err error) {
+			// Stop on error
+			if walkErr != nil {
+				return err
+			}
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/slicer/rowsreader/reader.go:50]: To sa tyka max 1% komponent, 99% komponent vytvori jeden CSV subor, bez gzip, zial, musime podporovat vsetky use-cases.

**Review by zajca — APPROVED** [review]

**Pull request merged**